### PR TITLE
Check quota of subdirectories when uploading to them

### PR DESCRIPTION
**Pull request by juliusknorr**
Different attempt to https://github.com/nextcloud/server/pull/23294 to check the available space on a subdirectory when uploading a file by dragging it to the browser.

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]
Code make sense! Minimal legacy code changes too! :+1:

**Review by ChristophWurst — APPROVED** [review]
Tested and works. I can upload a file to an external sub directory while the user's root directory is full :+1:

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -1236,6 +1236,7 @@
 				mtime: parseInt($el.attr('data-mtime'), 10),
 				type: $el.attr('data-type'),
 				etag: $el.attr('data-etag'),
+				quotaAvailableBytes: $el.attr('data-quota'),
```
Comment (ChristophWurst) [inline apps/files/js/filelist.js:1239]: this will also be set for files, but I guess it's fine

**Review by ChristophWurst — APPROVED** [review]
Still works after minor modifications :)

**Pull request merged**